commit 966d08028c8499db2c12c33a4de79ea261a689a6
Author: Friedemann Kleint <Friedemann.Kleint@digia.com>
Date:   Fri Nov 23 15:18:50 2012 +0100

    Draw disabled QMenu items in inactive state in vista style.
    
    Task-number: QTBUG-27992
    
    Change-Id: I4ed5661cfe0d0a10143224b73fbad6fac4927a37
    Reviewed-by: Joerg Bornemann <joerg.bornemann@digia.com>
    (cherry picked from qtbase/5223e7eb27dc15f8428aeb66dd8c57098dedce58)

diff --git a/src/gui/styles/qwindowsvistastyle.cpp b/src/gui/styles/qwindowsvistastyle.cpp
index ba47c87..7db6af3 100644
--- a/src/gui/styles/qwindowsvistastyle.cpp
+++ b/src/gui/styles/qwindowsvistastyle.cpp
@@ -1298,7 +1298,7 @@ void QWindowsVistaStyle::drawControl(ControlElement element, const QStyleOption
                                           menuitem->rect.y(), checkcol - 6, menuitem->rect.height()));
 
             if (act) {
-                stateId = MBI_HOT;
+                stateId = dis ? MBI_DISABLED : MBI_HOT;
                 XPThemeData theme2(widget, painter, QLatin1String("MENU"), MENU_POPUPITEM, stateId, option->rect);
                 d->drawBackground(theme2);
             }
